### PR TITLE
Travis integration

### PR DESCRIPTION
**Pull request by ftalbrecht**
This will go into master once we are finished with #66.

### CODE REVIEW

**Comment by ftalbrecht** [comment]
We are running into the 50 minute limit of travis. Any idea to speed things up, @dune-community/dune-gdt-devs?

**Comment by ftalbrecht** [comment]
I guess we could try to install the core modules via apt-get, perhaps this will buy us enough time for the tests to complete...

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  #- for ii in http://ftp.de.debian.org/debian/pool/main/u/ug/libug3.11.0_3.11.0-1_amd64.deb http://ftp.de.debian.org/debian/pool/main/u/ug/libug-dev_3.11.0-1_amd64.deb; do wget $ii; done && sudo -E dpkg -i *.deb && rm *.deb
+  - for ii in libdune-grid-2.3.1_2.3.1-1_amd64.deb libdune-grid-dev_2.3.1-1_amd64.deb; do wget http://de.archive.ubuntu.com/ubuntu/pool/universe/d/dune-grid/$ii; done && sudo -E dpkg --force-depends -i *.deb && rm *.deb
+  - wget http://de.archive.ubuntu.com/ubuntu/pool/universe/d/dune-istl/libdune-istl-dev_2.3.1-1_all.deb && sudo -E dpkg -i *.deb && rm *.deb
+  - wget http://de.archive.ubuntu.com/ubuntu/pool/universe/d/dune-localfunctions/libdune-localfunctions-dev_2.3.1-1_all.deb && sudo -E dpkg -i *.deb && rm *.deb
+  - for ii in http://de.archive.ubuntu.com/ubuntu/pool/universe/d/dune-typetree/libdune-typetree-2.3.1_2.3.1-1_amd64.deb http://de.archive.ubuntu.com/ubuntu/pool/universe/d/dune-typetree/libdune-typetree-dev_2.3.1-1_amd64.deb; do wget $ii; done && sudo -E dpkg -i *.deb && rm *.deb
+  - test -d src || git clone https://github.com/dune-community/dune-gdt-super.git src
+  - cd $HOME/src
+  - git checkout refactor_operators_functionals
```
Comment (renemilk) [inline .travis.yml]: ```git clone --recurse-submodules --depth=1 --branch=refactor_operators_functionals```

**Comment by renemilk** [comment]
AFAICT it's not building the tests that's an issue, since the tests do actually start to run. Can you list how long each test runs on your local machine? Then, instead of calling make test, you can use the ```cmake -R mask``` mechanism to deselect that and the negative mask in another builder.

**Comment by ftalbrecht** [comment]
For future reference, @renemilk : the missing configuration error in travis usually originates from a broken `.travis.yml`, see https://github.com/travis-ci/travis-ci/issues/2025.

**Comment by ftalbrecht** [comment]
Also for future reference: once we have a `.travis.yml` in master it supposedly helps (as in: we get an error message on the syntax error, including which line) to disable `Build only if .travis.yml is present` here: https://travis-ci.org/dune-community/dune-gdt/settings

**Comment by renemilk** [comment]
I should have remembered that yeah, sorry. I did run into malformed yaml myself some time ago. There's an online linter and the travis gem can do offline linting of the config file IIRC. Maybe add that to the pre-push hook.

**Comment by ftalbrecht** [comment]
The remaining issues are actual errors in code, the travis integration seems to work fine. For future reference: there currently exist the test "categories"

* operator
* projection
* prolongation
* space
* linearelliptic_cg_alugrid
* linearelliptic_cg_sgrid
* hyperbolic_fv_yaspgrid
* headercheck

If we add a new test binary which does not contain one of these strings we need to define a new category (aka `test_binaries_...` target, see `dune/gdt/test/CMakeLists.txt`) and add a new travis builder (see `.travis.yml`).

**Pull request merged**